Core/Src/main.c

```diff
@@ -82,7 +96,24 @@ int main(void)
      * 7. Full regression test as much as possible. If don't know what
      * regression test is, read up on it.
      */
-
+	HAL_StatusTypeDef ret;
+	uint8_t hum_val[3]; //humidity raw value
+	uint8_t temp_val[3]; //temperature raw value
+	typedef int32_t BME280_S32_t;
+	typedef uint32_t BME280_U32_t;
+	BME280_S32_t t_fine, adc_T, adc_H;
+	uint8_t spi_buff[5]; //SPI buffer not for measurements
+	uint8_t hum_osrs = 0b00000001; //enable humidity with x1 oversampling
+	uint8_t meas_val = 0b00100001; //set forced mode and turn off press
+	uint8_t filter_off = 0b00000000; //turn off filter
+	BME280_U32_t humidity; //final humidity value
+	uint8_t digT1[2];
+	int8_t digT2[2], digT3[2], digH2[2];

```

Comment:
I would count these as magic numbers as well. 
And I don't understand why you're mixing formats when naming variables. 
filter_off, BME280_U32_t, digT2.
What we've been using is snake_case for variables and camelCase for functions.